### PR TITLE
provisioned notification

### PR DESCRIPTION
**Pull request by tinakurian**
e2e-tests related to notification creation in MUR: https://github.com/codeready-toolchain/host-operator/pull/203

Jira: https://issues.redhat.com/browse/CRT-509

### CODE REVIEW

File: test/e2e/e2e_test.go [hunk 1]
```diff
@@ -325,9 +325,17 @@ func verifyResourcesProvisionedForSignup(t *testing.T, awaitility *wait.Awaitili
 		UserAccountStatus: userAccount.Status,
 	}
 	_, err = hostAwait.WaitForMasterUserRecord(mur.Name,
-		wait.UntilMasterUserRecordHasConditions(provisioned()),
+		wait.UntilMasterUserRecordHasConditions(provisioned(), provisionedNotificationCRCreated()),
 		wait.UntilMasterUserRecordHasUserAccountStatuses(expectedEmbeddedUaStatus))
 	assert.NoError(t, err)
+
+	notification, err := hostAwait.WaitForNotification(userAccount.Name + "-provisioned")
+	assert.NoError(t, err)
```
Comment (alexeykazakov) [inline test/e2e/e2e_test.go:333]:
```suggestion
	require.NoError(t, err)
```

File: test/e2e/e2e_test.go [hunk 2]
```diff
@@ -325,9 +325,17 @@ func verifyResourcesProvisionedForSignup(t *testing.T, awaitility *wait.Awaitili
...
+	require.NotNil(t, notification)
+	assert.Equal(t, notification.Name, userAccount.Name+"-provisioned")
+	assert.Equal(t, notification.Namespace, mur.Namespace)
+	assert.Equal(t, notification.Spec.Template, "userprovisioned")
+	assert.Equal(t, notification.Spec.UserID, userAccount.Spec.UserID)
```
Comment (alexeykazakov) [inline test/e2e/e2e_test.go:338]: The first param is the expected value. The second is the actual one. Please swap them.

File: wait/host.go [hunk 3]
```diff
@@ -55,6 +55,27 @@ func (a *HostAwaitility) WaitForMasterUserRecord(name string, criteria ...Master
...
+	err := wait.Poll(a.RetryInterval, a.Timeout, func() (done bool, err error) {
+		notification = &toolchainv1alpha1.Notification{}
+		if err := a.Client.Get(context.TODO(), types.NamespacedName{Namespace: a.Ns, Name: name}, notification); err != nil {
+			if errors.IsNotFound(err) {
+				a.T.Logf("waiting for availability of notification '%s'", name)
+				return false, nil
+			}
+			return false, err
+		}
+		if notification.Name != "" && notification.Spec.UserID != "" {
```
Comment (alexeykazakov) [inline wait/host.go:70]: Why do we need that check? notification.Name and Spec.UserID can't be nil here, right?

Comment (tinakurian) [reply wait/host.go:70]: @alexeykazakov sorry, this wasn't really ready for review, but i've changed the PR to open. The tests are failing for me which is why i wasn't going to open it yet lol. But i will address these comments soon

File: test/e2e/e2e_test.go (hunk 1 above)
Comment (tinakurian) [reply test/e2e/e2e_test.go]: fixed as part of: 4afd56d532ea1db81994c09a35511c8bc26e2b79

File: test/e2e/e2e_test.go (hunk 2 above)
Comment (tinakurian) [reply test/e2e/e2e_test.go]: fixed as part of: 4afd56d532ea1db81994c09a35511c8bc26e2b79

File: wait/host.go (hunk 3 above)
Comment (tinakurian) [reply wait/host.go]: fixed here: 46d9849d6db6dfe9991c5a66f4a3e53347166d56

**Pull request merged**